server/src/test/java/org/elasticsearch/indices/recovery/RecoverySourceHandlerTests.java

```diff
@@ -126,6 +129,7 @@
 
     private ThreadPool threadPool;
     private Executor recoveryExecutor;
+    private RecoveryPlannerService recoveryPlannerService;
```

Comment:
We're changing the `RecoverySourceHandler` in interesting ways. I realise it's mostly covered by other higher-level tests, but it'd be great if we could add unit tests for these changes to this suite somehow. At least checking somehow that we're calling the planner service as expected.